app/fileManager.js

```diff
@@ -68,6 +82,8 @@ export default function(state, emitter) {
       if (!ok) {
```

Comment:
It's wasteful to make two requests to the server. Try to remove the `exists` call and use the result of `getDLCounts` to also determine if the file no longer exists.